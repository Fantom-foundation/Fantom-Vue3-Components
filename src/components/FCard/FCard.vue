<template>
    <Component :is="tag" class="fcard">
        <slot></slot>
    </Component>
</template>

<script>
export default {
    name: 'FCard',

    props: {
        /** Tag name of the root element */
        tag: {
            type: String,
            default: 'div',
        },
    },
};
</script>

<style lang="scss">
@use 'style';
</style>
