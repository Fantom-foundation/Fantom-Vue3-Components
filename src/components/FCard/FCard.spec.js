import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import FCard from '@/components/FCard/FCard.vue';

let wrapper = null;

function createWrapper({ propsData = {}, slots = {}, parentComponent = null } = {}) {
    wrapper = null;

    wrapper = mount(FCard, {
        propsData,
        slots,
        parentComponent,
    });
}

describe('FCard', () => {
    it('should render correctly', () => {
        createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render content in default slot', () => {
        createWrapper({
            slots: {
                default: ['<span>content</span>'],
            },
        });

        expect(wrapper.html()).toContain('<span>content</span>');
    });
});
