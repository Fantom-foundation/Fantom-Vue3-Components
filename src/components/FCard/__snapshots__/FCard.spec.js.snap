// Vitest Snapshot v1

exports[`FCard > should render correctly 1`] = `
<div
  class="fcard"
>
  
  
</div>
`;
